sds_data_manager/lambda_code/efs_lambda/lambda_function.py

```diff
@@ -0,0 +1,133 @@
+import logging
+import os
+from pathlib import Path
+
+import boto3
+
+# Define the paths
+mount_path = os.environ.get("EFS_MOUNT_PATH")
+
+attitude_symlink_path = f"{mount_path}/latest_attitude_kernel.ah.a"
+ephemeris_symlink_path = f"{mount_path}/latest_ephemeris_kernel.bsp"
+
+
+def create_symlink(source_path: str, destination_path: str) -> None:
+    """Create a symlink from source_path to destination_path
+
+    Parameters
+    ----------
+    source_path : str
+        Source path of the symlink
+    destination_path : str
+        Destination path of the symlink
+    """
+
+    # Check if the symlink exists
+    if Path(destination_path).is_symlink():
+        # Remove the old symlink
+        Path(destination_path).unlink(missing_ok=True)
+
+    # Create a new symlink pointing to the new file
+    dest_symlink = Path(destination_path)
+    dest_symlink.symlink_to(source_path)
+
+
+def write_data_to_efs(s3_key: str, s3_bucket: str):
+    """Write data to EFS and create/update symlink
+
+    Parameters
+    ----------
+    s3_key : str
+        S3 object key
+    s3_bucket : str
+    """
+    filename = os.path.basename(s3_key)
+
+    # Create an S3 client
+    s3_client = boto3.client("s3")
+
+    # Download the file to the mount directory. Eg. /mnt/efs
+    destination_path = f"{mount_path}/" + os.path.basename(s3_key)
```

Comment:
you can re-use the filename from above.
```suggestion
    destination_path = f"{mount_path}/" + filename
```
This could also be `mount_path / filename` if you make it a Path object above.